File: 2015/c/01.c
// Advent of Code 2015, day 1
// (c) aichingert

#include <stdio.h>
#include <stdlib.h>

int main(void) 
{
	FILE *fp = fopen("../input/01", "r");
	char ch;
	int loc = 0;
	int part = 1;
	int iter = 0;

	while ((ch = fgetc(fp)) != EOF)
	{
		if (part) 
		{
			iter += 1;
		}

		if (ch == '(') 
		{
			loc += 1;
		} else if (ch == ')') 
		{
			loc -= 1;
		}

		if (part && loc < 0) 
		{
			part = 0;
		}
	}

	printf("Part 1: %i\n", loc);
	printf("Part 2: %i\n", iter);

	fclose(fp);
	return 0;
}

File: 2015/julia/01.jl
# Advent of Code 2015, day 1
# (c) aichingert

open("../input/01","r") do file
    i = 0
    d = 0
    p2 = 0

    while !eof(file)
        ch = read(file, Char)
        i += 1

        if ch == '(' d += 1 else d -= 1 end

        if d < 0 && p2 == 0 p2 = i end
    end

    println("Part 1: " * string(d))
    println("Part 2: " * string(p2))
end
